linkerd/app/outbound/src/http/breaker/consecutive_failures.rs

```diff
@@ -97,8 +94,8 @@ impl ConsecutiveFailures {
     /// Wait for a response to determine whether the breaker should be opened.
     async fn probation(&mut self) -> Result<classify::Class, ()> {
         tracing::debug!("Probation");
-        self.semaphore.add_permits(1);
-        self.gate.limit(self.semaphore.clone());
+        let sem = self.gate.limit().map_err(|_| ())?;
+        sem.add_permits(1);
```

Comment:
one thing i noticed here is that this change does mean that when the gate transitions to the limited state, the permits are not already present in the semaphore, so another task may be woken by the state transition, and then immediately yield again when it tries to actually wait on the semaphore. i don't think this is an actual correctness issue, since we will immediately re-wake that task by adding the permit, but it might be slightly less efficient, since we could do an unneeded round-trip through the scheduler.

we might want to have the `limit` method take a number of permits and create the new semaphore with that number of permits, so that when we actually transition the state, the permits are already present? 

not a big deal either way, just a thought.